### PR TITLE
@foxglove/ws-protocol: Add support for sessionId field

### PR DESCRIPTION
**Pull request by achim-k**
**Public-Facing Changes**
- @foxglove/ws-protocol: Add support for sessionId field

**Description**
See title

### CODE REVIEW

**Comment by jhurliman** [comment]
@achim-k, what do you think about just filling out `sessionId` by default? Use `+new Date()` for example.

File: typescript/ws-protocol/src/FoxgloveServer.ts [hunk 1]
```diff
@@ -323,6 +331,8 @@ export default class FoxgloveServer {
       name: this.name,
       capabilities: this.capabilities,
       supportedEncodings: this.supportedEncodings,
+      metadata: this.metadata,
+      sessionId: this.sessionId,
```
Comment (achim-k) [inline typescript/ws-protocol/src/FoxgloveServer.ts:335]:
```suggestion
      sessionId: this.sessionId ?? `${new Date().toUTCString()}`,
```

:point_up: @jhurliman is this what you mean?

Comment (jhurliman) [reply typescript/ws-protocol/src/FoxgloveServer.ts:335]: Yeah either way, I just think having `sessionId` set automatically is a good behavior.

**Review by jhurliman — APPROVED** [review]

File: typescript/ws-protocol/src/FoxgloveServer.ts [hunk 2]
```diff
@@ -99,14 +101,20 @@ export default class FoxgloveServer {
...
     capabilities?: string[];
     supportedEncodings?: string[];
+    metadata?: Record<string, string>;
+    sessionId?: string;
   }) {
     this.name = name;
     this.capabilities = capabilities ?? [];
     this.supportedEncodings = supportedEncodings;
+    this.metadata = metadata;
+    this.sessionId = sessionId;
```
Comment (achim-k) [inline typescript/ws-protocol/src/FoxgloveServer.ts:117]:
```suggestion
    this.sessionId = sessionId ?? `${new Date().toUTCString()}`;
```

**Pull request merged**